pkg/middleware/middleware.go

```diff
@@ -37,6 +40,19 @@ func LogRequests(w io.Writer) Middleware {
 	}
 }
 
+func RequestID(generator RequestIDGenerator, key string, isContext bool) Middleware {
+	return func(_ http.ResponseWriter, r *http.Request) *http.Request {
+		if isContext {
+			ctx := context.WithValue(r.Context(), key, generator())
```

Comment:
Please create a unique type for the context key, like `type requestIdKey string`, and wrap `key` with that type by doing `requestIdKey(key)` here. Use of raw strings as context keys is discouraged for collision reasons.

Reply:
> Please create a unique type for the context key, like `type requestIdKey string`, and wrap `key` with that type by doing `requestIdKey(key)` here. Use of raw strings as context keys is discouraged for collision reasons.

oh i didn't know about that i will do that thanks